frontend/src/metabase/admin/performance/components/StrategyEditorForDatabases.tsx

```diff
@@ -0,0 +1,379 @@
+import type { Location } from "history";
```

Comment:
This file is really long. Breaking ~200 LoC is when I personal start thinking about code getting split out to another file, 250 is my limit unless there's an obviously good reason for it.

Reply:
I've moved a bunch of logic into separate hooks and components, so the file is now down to 223 LOC, and is more readable. Big thanks to you for the suggestion.